src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -0,0 +1,81 @@
+module.exports = {
+  beforeCreate(event) {
+    console.log("**** beforeCreate hook is activated! ****");
+    const [eventTests, eventMetaTests] = [
+      event.params.data.tests,
+      event.params.data.MetaTest,
+    ];
+    runMetaTests(eventTests, eventMetaTests);
+  },
+
+  beforeUpdate(event) {
+    console.log("**** beforeUpdate hook is activated! ****");
+    const [eventTests, eventMetaTests] = [
+      event.params.data.tests,
+      event.params.data.MetaTest,
+    ];
+    runMetaTests(eventTests, eventMetaTests);
+  },
+};
+
+// runMetaTestsNeed more specific name later, maybe use "validateInternalTests"
+async function runMetaTests(eventTests, eventMetaTests) {
+  const [tests, metaTests] = await Promise.all([
+    getInternalTests(eventTests),
+    getMetaTests(eventMetaTests),
+  ]);
+
+  tests.map((test) =>
+    console.log("final test.internalTest: ", test.internalTest)
+  );
+
+  metaTests.map((metaTest) =>
+    console.log("final metaTest caseCode: ", metaTest.caseCode)
+  );
+
+  return false;
+}
+
+async function getInternalTests(eventTests) {
+  const internalTests = await strapi.db
+    .query("challenge.code-challenge-test")
+    .findMany({
+      select: ["id", "internalTest"],
+      where: {
+        id: getCurrentTestIDs(eventTests),
```

Comment:
getCurrentINTERNALTestIds

Reply:
Should I prefer Ids to IDs? I'm ok with either one but just not sure.

I read this SO response [https://stackoverflow.com/questions/15526107/acronyms-in-camelcase/27172000](url)

Reply:
it's whatever. I think Ids is more common

Reply:
I agree with the top answer, that's how I do it I guess